### PR TITLE
katex: enable persistent macros

### PR DESCRIPTION
**Pull request by qnxor**
See https://katex.org/docs/api.html#persistent-macros

Katex allows using `\gdef, \def, \newcommand` and the likes. However, if some `\gdef\root{\sqrt}` is defined in some math expression then it won't be recognized in subsequent math expressions unless `katex.render()` is called as in this PR (in my view this should be default behavior in Katex, as it's expected standard Latex behavior, but alas).

Basically, this PR allows one to define macros at the top of a post, then reuse them as many times as they wish afterwards (should work for subsequent posts too, though I haven't tested that) e.g.:
```
$\gdef\mylim{\lim\limits_{n\to\infty}}$

Text...

$$
\mylim \ln(1+x^n)=0
$$

More text reusing the macro inline $\mylim c_n = 1$.

And so on...
```

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]
Looks good, thanks @qnxor!
